packages/router/src/grpc/view-protocol-server/nullifier-status.ts

```diff
@@ -29,5 +28,18 @@ export const handleNullifierStatusReq = async (
   const noteSpent = noteByNullifier ? noteByNullifier.heightSpent !== 0n : false;
   const swapSpent = swapByNullifier ? swapByNullifier.heightClaimed !== 0n : false;
 
-  return new NullifierStatusResponse({ spent: noteSpent || swapSpent });
+  if (noteSpent || swapSpent) {
+    return new NullifierStatusResponse({ spent: true });
+  } else if (!req.awaitDetection) {
+    return new NullifierStatusResponse({ spent: false });
+  }
+
+  // Wait until our DB encounters a new note with this nullifier
+  const subscription = indexedDb.subscribe('SPENDABLE_NOTES');
+  for await (const update of subscription) {
+    const note = SpendableNoteRecord.fromJson(update.value);
+    if (note.nullifier?.equals(req.nullifier)) break;
+  }
+
+  return new NullifierStatusResponse({ spent: true });
```

Comment:
With `awaitDetection`, we should only return a response when the nullifier is spent, not when it is detected.

Reply:
Oh! Good catch. Will fix.